main.tf

```diff
@@ -91,9 +91,10 @@ resource "yandex_kubernetes_cluster" "main" {
     master_logging {
       enabled                    = var.master_logging["enabled"]
       log_group_id               = var.master_logging["log_group_id"] != "" ? var.master_logging["log_group_id"] : (var.master_logging["create_log_group"] ? yandex_logging_group.main[0].id : null)
-      kube_apiserver_enabled     = var.master_logging["kube_apiserver_enabled"]
-      cluster_autoscaler_enabled = var.master_logging["cluster_autoscaler_enabled"]
-      events_enabled             = var.master_logging["events_enabled"]
+      audit_enabled              = var.master_logging["enabled"] == false ? null : var.master_logging["audit_enabled"]
```

Comment:
maybe it will be better to simplify the construction to `var.master_logging["enabled"]  ? var.master_logging["audit_enabled"] : null`

wdyt?